### PR TITLE
Refactored AssignStudentToCourseCommand & Implemented CompositeID & builder.

### PR DESCRIPTION
**Pull request by ivanteejj**

### CODE REVIEW

**Comment by ivanteejj** [comment]
The main change done is as follows
1) as Progress does not have a unique key on its own, a compositeID, eg, (sid, aid), is created so as to be able to uniquely identify each Progress object. A compositeID is created using a compositeID builder and must contain at least 2 IDs for it to be considered a valid compositeID.
This works because each assignment can only be assigned to 1 and only 1 course.
The relevant tests have been written to ensure this.

2) Allow us to search and access for Students and Course by ID, via the model manager. e.g. has(ID x) and get(ID y)

3) Significantly refactor the assignStudentToCourse command's execute portion by utilizing the 
2)'s functionality. 

4) Added a Progress AB and Progress objects

File: src/main/java/seedu/address/logic/commands/commandAssign/AssignStudentToCourseCommand.java [hunk 1]
```diff
@@ -42,51 +43,39 @@ public static boolean isValidDescriptor(AssignDescriptor assignDescriptor) {
...
         } else {
-            ID courseid = ParserUtil.parseCourseid(courseidString);
-            ID studentid = ParserUtil.parseStudentid(studentidString);
-            foundCourse.addStudent(studentid);
-            foundStudent.addCourse(courseid);
+            Course foundCourse = model.getCourse(courseID);
+            Student foundStudent = model.getStudent(studentID);
+
+            foundCourse.addStudent(studentID);
+            foundStudent.addCourse(courseID);
```
Comment (h12567) [inline src/main/java/seedu/address/logic/commands/commandAssign/AssignStudentToCourseCommand.java:70]: why do we still need this? I thought we center the data to Course class

Comment (h12567) [reply src/main/java/seedu/address/logic/commands/commandAssign/AssignStudentToCourseCommand.java:70]: So it means no need to add to student class

**Review by h12567 — COMMENTED** [review]
> The main change done is as follows
> 
> 1. as Progress does not have a unique key on its own, a compositeID, eg, (sid, aid), is created so as to be able to uniquely identify each Progress object. A compositeID is created using a compositeID builder and must contain at least 2 IDs for it to be considered a valid compositeID.
>    This works because each assignment can only be assigned to 1 and only 1 course.
>    The relevant tests have been written to ensure this.
> 2. Allow us to search and access for Students and Course by ID, via the model manager. e.g. has(ID x) and get(ID y)
> 3. Significantly refactor the assignStudentToCourse command's execute portion by utilizing the
>    2)'s functionality.
> 4. Added a Progress AB and Progress objects

sounds good to me

File: src/main/java/seedu/address/logic/commands/commandAssign/AssignStudentToCourseCommand.java (hunk 1 above)
Comment (ivanteejj) [reply src/main/java/seedu/address/logic/commands/commandAssign/AssignStudentToCourseCommand.java:70]: Okay, the thing is I havent' finished implementing this part. Second thing is I thought really long about it and I think we should have a list of courseID in student. Please see here for diagram

photo6060082817980803656

**Review by ShengXue97 — APPROVED** [review]
Nice!

**Review by tiendat161 — APPROVED** [review]

**Pull request merged**